### ISSUE TITLE
index.html

### ISSUE THREAD

**Issue by t2tz** [t2tz/tanstack-template#1]
<?php
include 'jdf.php';

// تابع برای گرفتن قیمت‌ها از سایت tgju
function getPrices() {
    $base = "https://call3.tgju.org/ajax.json";
    $curlSession = curl_init();
    curl_setopt($curlSession, CURLOPT_USERAGENT, "User-Agent: Mozilla/5.0");
    curl_setopt($curlSession, CURLOPT_REFERER, 'https://tgju.org');
    curl_setopt($curlSession, CURLOPT_URL, $base);
    curl_setopt($curlSession, CURLOPT_HEADER, 0);
    curl_setopt($curlSession, CURLOPT_TIMEOUT, 60);
    curl_setopt($curlSession, CURLOPT_RETURNTRANSFER, 1);
    curl_setopt($curlSession, CURLOPT_FOLLOWLOCATION, true);
    $response = curl_exec($curlSession);

    if (curl_errno($curlSession)) {
        echo 'خطا: ' . curl_error($curlSession);
    }

    curl_close($curlSession);

    return json_decode($response, true)['current'] ?? [];
}

// تابع برای گرفتن قیمت بیت کوین از Binance
function getBitcoinPrice() {
    $url = "https://api.binance.com/api/v3/ticker/price?symbol=BTCUSDT";
    $result = file_get_contents($url);

    if ($result === FALSE) {
        return null;
    }

    $response = json_decode($result, true);
    return $response['price'] ?? null;
}

if (isset($_GET['ajax'])) {
    $prices = getPrices();
    $bitcoin_price = getBitcoinPrice();
    if ($bitcoin_price !== null) {
        $prices['crypto-bitcoin-usd'] = ['p' => $bitcoin_price];
    }

    echo json_encode($prices);
    exit;
}
?>

<!DOCTYPE html>
<html lang="fa">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
    <title>قیمت بیت‌کوین</title>
    <style>
        body {
            font-family: Tahoma, Arial, sans-serif;
            direction: rtl;
            text-align: center;
            background-color: #000;
            color: #fff;
            margin: 0;
            padding: 0;
        }
        .container {
            display: flex;
            flex-direction: column;
            align-items: center;
            margin: 20px;
        }
        table {
            border-collapse: collapse;
            background: #1a1a1a;
            border: 1px solid #444;
        }
        td {
            padding: 10px;
            border: 1px solid #555;
            text-align: center;
        }
        .currency-name {
            font-size: 12px;
            font-weight: bold;
            width: 60px; /* عرض جداگانه برای نام ارز */
        }
        .price-value {
            font-size: 21px;
            width: 280px; /* عرض جداگانه برای لیست قیمت */
        }
        .price-change {
            font-size: 20px;
            width: 40px; /* عرض جداگانه برای ایموجی قیمت */
        }
        tr:nth-child(even) {
            background-color: #222;
        }
        tr:nth-child(odd) {
            background-color: #1a1a1a;
        }
        #update-time {
            margin-top: 8px;
            font-size: 10px;
            color: #ddd;
        }
        .price-increase {
            color: #28a745;
        }
        .price-decrease {
            color: #dc3545;
        }
        .price-stable {
            color: #ffc107;
        }
        #update-label {
            font-weight: bold;
            font-size: 14px;
        }
        #update-clock {
            font-size: 30px; /* تغییر سایز فونت ساعت */
            color: #ccc;
            letter-spacing: 4px; /* فاصله بین ساعت، دقیقه و ثانیه */
        }
        .clock {
            font-size: 16px; /* سایز فونت ساعت */
            color: #ccc; /* رنگ ساعت */
        }
    </style>
</head>
<body>

    <div class="container">
        <table>
            <tbody id="price-table">
                <tr>
                    <td class="currency-name">دلار</td>
                    <td id="price-dollar" class="price-value">...</td>
                    <td id="change-dollar" class="price-change">⚪</td>
                </tr>
                <tr>
                    <td class="currency-name">تتر</td>
                    <td id="price-tether" class="price-value">...</td>
                    <td id="change-tether" class="price-change">⚪</td>
                </tr>
                <tr>
                    <td class="currency-name">بیت کوین</td>
                    <td id="price-bitcoin" class="price-value">...</td>
                    <td id="change-bitcoin" class="price-change">⚪</td>
                </tr>
            </tbody>
        </table>
    </div>

 
    <p id="update-clock" class="clock">...</p>

    <script>
        let previousPrices = {};

        function fetchPrices() {
            fetch('?ajax=true')
                .then(response => response.json())
                .then(data => {
                    const items = {
                        'price_dollar_rl': { id: 'price-dollar', changeId: 'change-dollar' },
                        'crypto-tether-irr': { id: 'price-tether', changeId: 'change-tether' },
                        'crypto-bitcoin-usd': { id: 'price-bitcoin', changeId: 'change-bitcoin' }
                    };

                    for (const [key, item] of Object.entries(items)) {
                        if (data[key]?.p) {
                            const currentPrice = parseFloat(data[key].p.replace(/,/g, ''));
                            const previousPrice = previousPrices[key] || currentPrice;
                            previousPrices[key] = currentPrice;

                            let changeClass = 'price-stable';
                            let changeSymbol = '⚪';

                            if (currentPrice > previousPrice) {
                                changeClass = 'price-increase';
                                changeSymbol = '🟢';
                            } else if (currentPrice < previousPrice) {
                                changeClass = 'price-decrease';
                                changeSymbol = '🔻';
                            }

                            const unit = (key === 'crypto-bitcoin-usd') ? 'USD' : 'IRR';
                            document.getElementById(item.id).textContent = currentPrice.toLocaleString('en-US') + ' ' + unit;
                            document.getElementById(item.id).className = `price-value ${changeClass}`;
                            document.getElementById(item.changeId).textContent = changeSymbol;
                        }
                    }

                    document.getElementById('update-clock').textContent = new Date().toLocaleTimeString('fa-IR');
                })
                .catch(error => {
                    console.error('خطا در دریافت قیمت:', error);
                });
        }

        fetchPrices();
        setInterval(fetchPrices, 1000);
    </script>
</body>
</html>
